include/bitcoin/client/obelisk/obelisk_types.hpp

```diff
@@ -48,8 +40,10 @@ typedef std::vector<history_row> history_list;
 
 struct BCC_API stealth_row
 {
-    data_chunk ephemkey;
-    payment_address address;
+    // No sign byte in public key.
+    hash_digest ephemkey;
+    // No version byte in address.
+    short_hash address;
     hash_digest transaction_hash;
```

Comment:
It would be preferable to not change this structure. The problem with both changes is that the values are weakly-typed as hashes whereas the data represents a public key and a payment address. The wire optimizations don't need to impact the logical model exposed by this struct. We should be able to fabricate a proper `ec_public` and `payment_address` by simply adding the presumed bytes before populating this struct. Is it true that the version byte of the address can be set by convention?